### PR TITLE
[SLTM-1230 + SLTM-1231] fix head and delete requests 

### PR DESCRIPTION
**Pull request by veronikaKochugova**

### CODE REVIEW

File: src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java [hunk 1]
```diff
@@ -394,7 +394,21 @@ public final boolean await(final long timeout, final TimeUnit timeUnit)
 							.map(
 											stepSlice -> {
 												try {
-													return stepSlice.await(timeout, timeUnit);
+													var invokeTimeMillis = System.currentTimeMillis();
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java]: I'd consider it `final`

File: src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java [hunk 2]
```diff
@@ -394,7 +394,21 @@ public final boolean await(final long timeout, final TimeUnit timeUnit)
...
+													var timeOutMillis = timeUnit.toMillis(timeout);
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java]: this also

File: src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java [hunk 3]
```diff
@@ -394,7 +394,21 @@ public final boolean await(final long timeout, final TimeUnit timeUnit)
...
+													var await_result = false;
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java]: 1. Please use camelCase in java
2. Let's name it `sliceAwaitResult`

File: src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java [hunk 4]
```diff
@@ -394,7 +394,21 @@ public final boolean await(final long timeout, final TimeUnit timeUnit)
...
+													long elapsedTimeMillis;
+													while(timeOutMillis > (elapsedTimeMillis = System.currentTimeMillis() - invokeTimeMillis)) {
+														elapsedTimeMillis = System.currentTimeMillis() - invokeTimeMillis;
+														if (timeOutMillis <= elapsedTimeMillis) {
+															break;
+														}
+														if (Thread.currentThread().isInterrupted()){
+															throwUnchecked(new InterruptedException());
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java:407]: Isn't it enough to `break` the loop?

**Review by akurilov — CHANGES_REQUESTED** [review]

File: src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java (hunk 3 above)
Comment (veronikaKochugova) [reply src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java]: sorry, already used to use snake style on python scripts :)

File: src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java (hunk 4 above)
Comment (veronikaKochugova) [reply src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java:407]: Isn’t it better to see a stacktrace, as is the case with stndalone mode?

Comment (akurilov) [reply src/main/java/com/emc/mongoose/base/load/step/client/LoadStepClientBase.java:407]: maybe

**Review by akurilov — APPROVED** [review]

**Review by akurilov — APPROVED** [review]

**Pull request merged**